setup.py

```diff
@@ -16,7 +16,7 @@
     py_modules=['eth_bloom'],
     setup_requires=['setuptools-markdown'],
     install_requires=[
-        "pysha3>=0.3",
+        "eth-hash[pycryptodome]>=0.1.*",
```

Comment:
What do you think about simply requiring `eth-hash` and having an `extras_require` for each of the hashing backends.  Since our *goal* is to be agnostic about what backend people choose i think we should avoid having any of the as hard requirements like this.

Probably deserves a small section in the README outlining that to use this library you **must** install one of the supported hashing backends.

Reply:
I like the idea, but let me go a step further. Basically, `pycryptodome` is not a requirement of `eth-bloom`, but rather of `eth-hash`. Shouldn't we instead just use `eth-hash` in `install_requires`, and not mention `pycryptodome` at all, since it is already mentioned in the `extras_require` of `eth-hash`? See: https://github.com/ethereum/eth-hash/blob/master/setup.py#L29

By mentioning `pycryptodome` in multiple projects, we will be tying together the release cycles of the Ethereum packages relying on `eth-hash`, if ever we need to bump the version of `pycryptodome`.

Reply:
I'm fine leaving the `pycryptodome` out completely, but I want to be sure that a user who naively installs `eth-bloom` without a hash backend encounters an error which *clearly* tells them what is wrong and how to fix it.

Reply:
Ok. After removing `pycryptodome` from `setup.py`, if you try to use `eth-bloom` without having a valid backend already installed, you get this:

```
>>> from eth_bloom import bloom
....
ImportError: None of these hashing backends are installed: ['pycryptodome', 'pysha3'].
Install with `pip install eth-hash[pycryptodome]`.
```

Seems to conform with your ideas, but, will this not complicate the installation of user-centric packages like web3.py and py-evm, since we will be doing similar changes across the Ethereum ecosystem? Normally, one would expect a fully functional system just by doing `pip install py-evm`.

If our goal is to let people choose whatever crypto backend they want, they can already do so by specifying any of the supported backends using the `ETH_HASH_BACKEND` environment variable.

I would therefore like you to reconsider shipping `pycryptodome` as the default backend, which is the current state of the PR.

Reply:
I have a *line* that I draw in my head where `trinity` and `web3.py` sit above that line and most everything else lives below it.  `web3.py` and `trinity` (which is the py-evm based ethereum node) will likely include one of the hash backends to ensure that the use experience remains the same.

Everything else (`eth-utils`, `py-trie`, `eth-bloom`, `eth-keys`, etc) will not make assumptions about what hashing backend you prefer and generally require you to pick.

I'm open to alternate suggestions, and I'm open to using `pycryptodome` as the default backend for `web3.py` and `trinity`.

👍 / 👎 ?

Reply:
Thanks for the clarification. I have incorporated your suggestions.